src/Buildalyzer/Logging/EventProcessor.cs

```diff
@@ -146,7 +146,8 @@ private void TargetFinished(object sender, TargetFinishedEventArgs e)
     private void MessageRaised(object sender, BuildMessageEventArgs e)
     {
         AnalyzerResult result = _currentResult.Count == 0 ? null : _currentResult.Peek();
-        if (result is object)
+        // do not process any sub project
+        if (result is object && (string.IsNullOrEmpty(result.Command) || AnalyzerManager.NormalizePath(e.ProjectFile) == _projectFilePath))

```

Comment:
Preferably, define a local function that does this, so that the comment is not needed due to the self explanatory name.

Reply:
done
Create a `IsRelevant' local function + method refactor to increase readibility